src/bootstrap.coffee

```diff
@@ -64,11 +73,15 @@ exchangeKey = ->
 		.then (device) ->
 			if not device?
 				throw new ExchangeKeyError("Couldn't fetch device with provisioning key")
-			# We found the device, we can try to generate a working device key for it
-			request.postAsync("#{config.apiEndpoint}/api-key/device/#{device.id}/device-key?apikey=#{userConfig.apiKey}")
+			# We found the device, we can try to register a working device key for it
+			userConfig.deviceApiKey ?= deviceRegister.generateUniqueKey()
+			request.postAsync("#{config.apiEndpoint}/api-key/device/#{device.id}/device-key?apikey=#{userConfig.apiKey}", { json: true, body: apiKey: userConfig.deviceApiKey })
```

Comment:
Can you split the options object onto new lines please, it's really easy to miss that we pass extra options atm as it's after a long string